### DISCUSSION TITLE
[Idea] Buildpack for any app that contains a Dockerfile

### DISCUSSION THREAD

**Discussion by collimarco** [buildpacks/community#145] [Ideas]
What do you think about having a generic buildpack that can pack/build any app that contains a Dockerfile?

Basically it would be a fallback, similar to a simple `docker build`, but having that **would make `pack` (and similar) more complete**.

Also, if in production you have some apps that use `pack` and other that use `docker build`, you could **always use the same CNB entrypoint**, without wondering every time how the image was built (pack / plain docker).

**Comment by dfreilich** [comment]
In general, the Buildpacks project doesn't maintain any buildpacks themselves; it's the different vendors who do. Of those, I believe that Google Cloud Run has a mechanism which operates like that, building if there is a Dockerfile present, though I'm not sure whether it works with a "Dockerfile" buildpack, or separately.

**Comment by dfreilich** [comment]
The idea has been raised in the past, but there has been pushback from many of the members of the project. I think that introducing that mechanism would complicate the workings of `pack` significantly. Ultimately, though, we'd welcome a contribution to the [Buildpacks Registry](https://registry.buildpacks.io/searches/Docker) of a Buildpack which did that!

**Comment by natalieparellano** [comment]
Possibly relevant: https://github.com/buildpacks/rfcs/blob/main/text/0105-dockerfiles.md introduces the concept of an image extension which could in theory consume a Dockerfile from an application directory.
